### DISCUSSION TITLE
Machine readable specification

### DISCUSSION THREAD

**Discussion by SuperAuguste** [build-server-protocol/build-server-protocol#485] [Q&A] [answered]
Hey, I'm looking to generate types for the build server protocol and was wondering if there are any plans to produce machine readable type definitions like the recent-ish [LSP meta-model](https://github.com/microsoft/vscode-languageserver-node/blob/main/protocol/metaModel.json). I'd be happy to explore this if you'd be interested.

**Comment by ckipp01** [comment, marked as answer]
Hey, great question @SuperAuguste. We're actually in the process of porting the the spec over to [Smithy](https://smithy.io/2.0/index.html) [here](https://github.com/build-server-protocol/build-server-protocol/pull/426). This will greatly help in situations like you're asking about as we'd be able to provide both the Smithy model and make it much easier to generate other things from this.
